### PR TITLE
Implement nanprod function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanprod, which computes the product of elements along a specified axis while ignoring NaN values in the input tensor.
The operation follows NumPy semantics for axis and keepdims and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/torch/numpy.py [hunk 1]
```diff
@@ -1318,6 +1318,25 @@ def nanmin(x, axis=None, keepdims=False):
...
 
 
+def nanprod(x, axis=None, keepdims=False):
+    x = convert_to_tensor(x)
+
+    if axis == () or axis == []:
+        return torch.nan_to_num(x, nan=1)
+
+    if isinstance(axis, (list, tuple)):
+        axis = sorted(axis, reverse=True)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py:1328]: There is no sorting on prod function.

File: keras/src/backend/numpy/numpy.py [hunk 2]
```diff
@@ -973,6 +973,19 @@ def nanmin(x, axis=None, keepdims=False):
...
+    axis = standardize_axis_for_numpy(axis)
+
+    x = convert_to_tensor(x)
+
+    dtype = dtypes.result_type(x.dtype)
+    if dtype in ("bool", "int8", "int16"):
+        dtype = "int32"
+    elif dtype in ("uint8", "uint16"):
+        dtype = "uint32"
+    return np.nanprod(x, axis=axis, keepdims=keepdims).astype(dtype)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py]: I now pass the target dtype to np.nanprod so the product is computed in the widened type and avoids integer overflow.

File: keras/src/backend/tensorflow/numpy.py [hunk 3]
```diff
@@ -2186,6 +2186,16 @@ def nanmin(x, axis=None, keepdims=False):
     )
 
 
+def nanprod(x, axis=None, keepdims=False):
+    x = convert_to_tensor(x)
+
+    if not x.dtype.is_floating:
+        return prod(x, axis=axis, keepdims=keepdims)
+
+    x_safe = tf.where(tf.math.is_nan(x), tf.ones_like(x), x)
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: nitpick: like for `nansum`, you don't need the full array of ones, just a scalar, so you can replace `tf.ones_like(x)` with `tf.ones((), dtype=dtype)`

File: keras/src/backend/torch/numpy.py [hunk 4]
```diff
@@ -1318,6 +1318,25 @@ def nanmin(x, axis=None, keepdims=False):
...
+
+    if not torch.is_floating_point(x):
+        return prod(x, axis=axis, keepdims=keepdims)
+
+    return prod(
+        torch.where(torch.isnan(x), torch.ones_like(x), x),
```
Comment (hertschuh) [inline keras/src/backend/torch/numpy.py]: Same here, use `torch.ones((), dtype=x.dtype)`.

**Review by hertschuh — COMMENTED** [review]
Thanks for adding this!

File: keras/src/backend/tensorflow/numpy.py (hunk 3 above)
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: Agreed. I’ve replaced tf.ones_like(x) with tf.ones((), dtype=x.dtype).

File: keras/src/backend/torch/numpy.py (hunk 4 above)
Comment (shashaka) [reply keras/src/backend/torch/numpy.py]: I updated.

**Review by hertschuh — APPROVED** [review]
Thanks!
